apis/oam/v1alpha2/register.go

```diff
@@ -0,0 +1,140 @@
+	ApplicationConfigurationGroupKind        = schema.GroupKind{Group: Group, Kind: ApplicationConfigurationKind}.String()
+	ApplicationConfigurationKindAPIVersion   = ApplicationConfigurationKind + "." + SchemeGroupVersion.String()
+	ApplicationConfigurationGroupVersionKind = SchemeGroupVersion.WithKind(ApplicationConfigurationKind)
+)
+
+// Server type metadata.
+var (
+	ServerKind             = reflect.TypeOf(Server{}).Name()
+	ServerGroupKind        = schema.GroupKind{Group: Group, Kind: ServerKind}.String()
+	ServerKindAPIVersion   = ServerKind + "." + SchemeGroupVersion.String()
+	ServerGroupVersionKind = SchemeGroupVersion.WithKind(ServerKind)
+)
+
+// SingletonServer type metadata.
+var (
+	SingletonServerKind             = reflect.TypeOf(SingletonServer{}).Name()
+	SingletonServerGroupKind        = schema.GroupKind{Group: Group, Kind: SingletonServerKind}.String()
+	SingletonServerKindAPIVersion   = SingletonServerKind + "." + SchemeGroupVersion.String()
+	SingletonServerGroupVersionKind = SchemeGroupVersion.WithKind(SingletonServerKind)
+)
+
+// Worker type metadata.
+var (
+	WorkerKind             = reflect.TypeOf(Worker{}).Name()
+	WorkerGroupKind        = schema.GroupKind{Group: Group, Kind: WorkerKind}.String()
+	WorkerKindAPIVersion   = WorkerKind + "." + SchemeGroupVersion.String()
+	WorkerGroupVersionKind = SchemeGroupVersion.WithKind(WorkerKind)
+)
+
+// SingletonWorker type metadata.
+var (
+	SingletonWorkerKind             = reflect.TypeOf(SingletonWorker{}).Name()
+	SingletonWorkerGroupKind        = schema.GroupKind{Group: Group, Kind: SingletonWorkerKind}.String()
+	SingletonWorkerKindAPIVersion   = SingletonWorkerKind + "." + SchemeGroupVersion.String()
+	SingletonWorkerGroupVersionKind = SchemeGroupVersion.WithKind(SingletonWorkerKind)
+)
+
+// Task type metadata.
+var (
+	TaskKind             = reflect.TypeOf(Task{}).Name()
+	TaskGroupKind        = schema.GroupKind{Group: Group, Kind: TaskKind}.String()
+	TaskKindAPIVersion   = TaskKind + "." + SchemeGroupVersion.String()
+	TaskGroupVersionKind = SchemeGroupVersion.WithKind(TaskKind)
+)
+
+// SingletonTask type metadata.
+var (
+	SingletonTaskKind             = reflect.TypeOf(SingletonTask{}).Name()
+	SingletonTaskGroupKind        = schema.GroupKind{Group: Group, Kind: SingletonTaskKind}.String()
+	SingletonTaskKindAPIVersion   = SingletonTaskKind + "." + SchemeGroupVersion.String()
+	SingletonTaskGroupVersionKind = SchemeGroupVersion.WithKind(SingletonTaskKind)
+)
+
+func init() {
+	SchemeBuilder.Register(&WorkloadDefinition{}, &WorkloadDefinitionList{})
+	SchemeBuilder.Register(&TraitDefinition{}, &TraitDefinitionList{})
+	SchemeBuilder.Register(&ScopeDefinition{}, &ScopeDefinitionList{})
+	SchemeBuilder.Register(&Component{}, &ComponentList{})
+	SchemeBuilder.Register(&ApplicationConfiguration{}, &ApplicationConfigurationList{})
+	SchemeBuilder.Register(&Server{}, &ServerList{})
```

Comment:
Also, is it possible to separate the core workload/trait/scope into another file/place? Conceptually, they are not the same type as the resources defined above.

Reply:
They should live in the same package because they're both the same API group and version. I've moved them into a separate file. I'd be happy to move them into a separate package if we'd be happy to change their API group - e.g. `workload.core.oam.dev` or similar.